### PR TITLE
feat(goTo): port to v3

### LINKED ISSUE #5732: [Feature Request] add horizontal support to scrolling features (v-scroll, goTo)
### Problem to solve
Vuetify scrolling tools: `v-scroll` directive and Scrolling motion `$vuetify.goTo()` does not work with horizontal elements

### Proposed solution
providing `v-horizontal-scroll` directive, add horizontal functionality to `$vuetify.goTo()` element, enhance mouse scroll movement over horizontal elements

codepen: https://codepen.io/anon/pen/jQKVOX

### LINKED ISSUE #13538: [Task] goto
# Overview

- Related Tasks: n/a
- Related Issues: n/a

A conversion should not start until *all* assigned issues with the milestone v2.2.x are resolved and in the `next` branch. Reference our [Coding Guidelines](https://github.com/vuetifyjs/vuetify/wiki/coding-guidelines) for information regarding the team's practices or reach out to us on [Discord](https://community.vuetifyjs.com).

### PR DESCRIPTION
**Pull request by johnleider**
## Motivation and Context

closes #13538
closes #5732

TODO:

- [x] Horizontal scrolling
- [x] RTL
- [x] Offset
- [x] Unit tests
- [x] Documentation

## Markup:
<details>

```vue
<template>
  <v-layout>
    <div ref="container" class="overflow-x-auto" style="width: 2200px;">
      <div
        id="test2"
        style="margin-left: 2000px; width: 200px;"
      >
        Testing 2

        <v-btn @click="goTo.horizontal(0, container)">Back to left</v-btn>
      </div>

      <v-main style="min-height: 700vh">
        <v-btn text="Vertical" @click="goTo('#test')" />
        <v-btn text="Horizontal" @click="goTo.horizontal(2000, container)" />

        <div id="test" ref="btn" style="margin-top: 300vh" />

        <v-btn @click="goTo(-100)">Back to Top</v-btn>
      </v-main>
    </div>
  </v-layout>
</template>

<script setup>
  import { ref } from 'vue'
  import { useGoTo } from '../src/composables/goto.ts'

  const btn = ref(null)
  const container = ref(null)
  const goTo = useGoTo()
</script>
```
</details>

### CODE REVIEW

File: packages/vuetify/src/composables/goto.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  let totalOffset = 0
+  while (el) {
+    totalOffset += horizontal ? el.scrollLeft : el.offsetTop
+    el = el.offsetParent as HTMLElement
+  }
+
+  return totalOffset
+}
+
+export function createGoTo (options: Partial<GoToOptions>, locale: LocaleInstance & RtlInstance) {
```
Comment (yuwu9145) [inline packages/vuetify/src/composables/goto.ts:71]: `Partial<GoToOptions> | undefined`


Screenshot 2024-01-16 at 7 12 08 pm

File: packages/vuetify/src/composables/goto.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  return scrollTo(_target, _container, options, true)
+}
+
+export function useGoTo (options?: Partial<GoToOptions>) {
+  const goTo = inject(GoToSymbol)
+
+  if (!goTo) throw new Error('[Vuetify] Could not find injected goto instance')
+
+  async function go (target: HTMLElement | string | number, container: HTMLElement | string | 'parent') {
+    return vertical(target, container, mergeDeep(options, goTo?.options) as GoToOptions)
```
Comment (yuwu9145) [inline packages/vuetify/src/composables/goto.ts:161]: reverse mergeDeep params? `mergeDeep(goTo?.options, options)`

As custom options from user's land `const goTo = useGoTo({ duration: 5000 })` gets override by default

**Comment by yuwu9145** [comment]
Tested, LGTM. Just a little bit optional typing improvement if you want #19057

File: packages/vuetify/src/composables/goto.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      reachEnd = clientSize + container.scrollLeft >= container.scrollWidth
+
+      if (targetLocation > container.scrollLeft && reachEnd) return resolve(targetLocation)
+    }
+
+    requestAnimationFrame(step)
+  }))
+}
+
+export function useGoTo (_options: Partial<GoToOptions> = {}) {
```
Comment (KaelWD) [inline packages/vuetify/src/composables/goto.ts:144]: Why is this a composable instead of just `import { goTo } from 'vuetify'`?

Comment (johnleider) [reply packages/vuetify/src/composables/goto.ts:144]: Future flexibility

**Pull request merged**